lib/azure/armrest/exception.rb

```diff
@@ -21,16 +32,28 @@ def message
     class ApiException < Exception
       attr_accessor :code
 
+      # Create a new ApiException class. The +code+ is the error code.
+      #
+      # This class serves as the parent
       def initialize(code, message, cause_exception)
         @code = code
         super(message, cause_exception)
       end
 
+      # A stringified version of the error. If self is a plain ApiException,
+      # then the cause is included to aid in debugging.
+      #
       def to_s
-        "[#{code}] #{message}"
+        if self.class == ApiException && cause
```

Comment:
Because you know the error based on the name, e.g. a TooManyRequests exception already tells me the cause - too many requests. The cause then becomes redundant. But, with ApiException it isn't immediately clear.